Sources/PrimerSDK/Classes/Core/Payment Services/ApplePayService.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  ApplePayService.swift
+//  PrimerSDK
+//
+//  Created by Evangelos Pittas on 16/4/21.
+//
+
+import Foundation
+
+
+protocol ApplePayServiceProtocol {
+    func tokenize(instrument: PaymentInstrument, completion: @escaping (Result<PaymentMethodToken, Error>) -> Void)
+    func fetchConfig(_ completion: @escaping (Error?) -> Void)
+}
+
+enum ApplePayType {
+    case recurring, checkout
+}
+
+class ApplePayService: NSObject, ApplePayServiceProtocol {
+    
+    deinit {
+        print("ApplePayService deinit")
+    }
+    
+    func fetchConfig(_ completion: @escaping (Error?) -> Void) {
+        let state: AppStateProtocol = DependencyContainer.resolve()
+        
+        guard let clientToken = state.decodedClientToken else {
+            return completion(PrimerError.configFetchFailed)
+        }
+        
+        let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+        
+        let api: PrimerAPIClientProtocol = DependencyContainer.resolve()
+
+        api.fetchConfiguration(clientToken: clientToken) { [weak self] (result) in
+            switch result {
+            case .failure(let error):
+                completion(error)
+            case .success(let config):
+                state.paymentMethodConfig = config
+
+                state.viewModels = []
+
+                config.paymentMethods?.forEach({ method in
+                    guard let type = method.type else { return }
+                    if type == .googlePay { return }
+                    state.viewModels.append(PaymentMethodViewModel(type: type))
+                })
+
+                // ensure Apple Pay is always first if present.
```

Comment:
[Q]
Why do you want to ensure this?

Reply:
This was there in the code before I join. I thought it's some kind of business decision that I wasn't aware of. @ceriksson may have an insight. Do you want me to remove it?

Reply:
this was a design request, something with Apple wanting Apple Pay to be displayed as the first option. Not sure whether this is a crucial requirement by Apple. If we want to I believe we can leave it out :)

Reply:
Oh yeah I remember. We can keep it like this but let's have a proper comment + something on the documentation about this info :)

Reply:
Cool, will add a comment